### PR TITLE
Fixes #96 by handling RepositoryNotFoundError

### LINKED ISSUE #96: RepositoryNotFoundError when running scripts/run_dpo.py with local repo
Steps to repro:

* Use HF alignment commit: https://github.com/huggingface/alignment-handbook/commit/98fe28fb14b6bacedc23bacb166bb23cc1bf1cc0
* In recipes/../dpo/config_full.yaml, use a local directory for the model, eg: `path/to/model`

It will throw an error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_errors.py", line 261, in hf_raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.10/dist-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://huggingface.co/api/models/data/zephyr-1b-sft-full/tree/main?recursive=True&expand=False

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/alignment-handbook/scripts/run_dpo.py", line 220, in <module>
    main()
  File "/alignment-handbook/scripts/run_dpo.py", line 121, in main
    if is_adapter_model(model, model_args.model_revision):
  File "/alignment-handbook/src/alignment/model_utils.py", line 102, in is_adapter_model
    repo_files = list_repo_files(model_name_or_path, revision=revision)
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_deprecation.py", line 101, in inner_f
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_validators.py", line 118, in _inner_fn
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/hf_api.py", line 2213, in list_repo_files
    return [
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/hf_api.py", line 2213, in <listcomp>
    return [
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/hf_api.py", line 2179, in list_files_info
    for subpath_info in paginate(path=tree_url, headers=headers, params={"recursive": True, "expand": expand}):
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_pagination.py", line 36, in paginate
    hf_raise_for_status(r)
  File "/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_errors.py", line 293, in hf_raise_for_status
    raise RepositoryNotFoundError(message, response) from e
huggingface_hub.utils._errors.RepositoryNotFoundError: 404 Client Error. (Request ID: Root=1-659ea868-2ceb4a3e51084810535b1d81;2c59fce0-fb33-4d89-ac5e-93006069031d)

Repository Not Found for url: https://huggingface.co/api/models/data/zephyr-1b-sft-full/tree/main?recursive=True&expand=False.
Please make sure you specified the correct `repo_id` and `repo_type`.
If you are trying to access a private or gated repo, make sure you are authenticated.
```

It looks like the problem is [this code in model_utils.py](https://github.com/huggingface/alignment-handbook/blob/f0ffa0d7a6ab666b1f80f3f7dbb3c6364ac31967/src/alignment/model_utils.py#L101-L108)

```
def is_adapter_model(model_name_or_path: str, revision: str = "main") -> bool:
    try:
        # Try first if model on a Hub repo
        repo_files = list_repo_files(model_name_or_path, revision=revision)
    except HFValidationError:
        # If not, check local repo
        repo_files = os.listdir(model_name_or_path)
    return "adapter_model.safetensors" in repo_files or "adapter_model.bin" in repo_files
```

Is not catching `RepositoryNotFoundError`

### PR DESCRIPTION
**Pull request by tleyden**
Fix https://github.com/huggingface/alignment-handbook/issues/96 by adding exception handling for `RepositoryNotFoundError` in addition to the existing handling for `HFValidationError`

### CODE REVIEW

File: src/alignment/model_utils.py [hunk 1]
```diff
@@ -105,6 +106,9 @@ def is_adapter_model(model_name_or_path: str, revision: str = "main") -> bool:
     except HFValidationError:
```
Comment (lewtun) [inline src/alignment/model_utils.py:106]:
```suggestion
    except (HFValidationError, RepositoryNotFoundError):
```

File: src/alignment/model_utils.py [hunk 2]
```diff
@@ -105,6 +106,9 @@ def is_adapter_model(model_name_or_path: str, revision: str = "main") -> bool:
...
         # If not, check local repo
         repo_files = os.listdir(model_name_or_path)
+    except RepositoryNotFoundError:
```
Comment (lewtun) [inline src/alignment/model_utils.py:109]: If I'm not mistaken we can gather this exception with the one above - would you mind checking that works?

**Review by lewtun — COMMENTED** [review]
Thanks for catching this extra exception @tleyden ! I left a suggestion to keep the code tidy but otherwise this looks good :)

File: src/alignment/model_utils.py (hunk 2 above)
Comment (tleyden) [reply src/alignment/model_utils.py:109]: Sounds good!  I'll run a quick sanity check

Comment (tleyden) [reply src/alignment/model_utils.py]: @lewtun I added a simple unit test.  If needed I could enhance the test with `unittest.mock`

**Review by lewtun — APPROVED** [review]
Thanks for iterating @tleyden ! Can you please run `make style && make quality` to ensure the tests pass?

**Comment by tleyden** [comment]
@lewtun I pushed another commit

**Comment by tleyden** [comment]
@ArthurZucker I saw your recent comment on another PR, so it looks like you are online.  Any chance you can approve kicking off the workflow on this PR?

**Comment by ArthurZucker** [comment]
Sure!

**Comment by lewtun** [comment]
Hey @tleyden it seems you need to reorder some imports in `model_utils.py` which `make style` should fix

**Comment by tleyden** [comment]
Thanks @lewtun will try to find some time today!

**Comment by tleyden** [comment]
@lewtun I pushed another commit, PTAL

**Comment by lewtun** [comment]
Thanks!

**Pull request merged**